### PR TITLE
calls: init at 0.0.1

### PR DESCRIPTION
**Pull request by craigem**
###### Motivation for this change

The calls application is required to provide phone calls for the Librem 5 phone.
    
This is the initial commit for this application to be packaged for NixOS and the first step to getting NixOS to fully support the Librem 5 phone.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by craigem** [comment]
Resolves #65361

File: pkgs/applications/radio/calls/default.nix
Comment (peterhoeg) [inline pkgs/applications/radio/calls/default.nix]: we can get rid of the ```let``` part

File: pkgs/applications/radio/calls/default.nix
Comment (peterhoeg) [inline pkgs/applications/radio/calls/default.nix]: just assign pname and version here (and drop name)

File: pkgs/applications/radio/calls/default.nix
Comment (peterhoeg) [inline pkgs/applications/radio/calls/default.nix]: not needed

File: pkgs/applications/radio/calls/default.nix
Comment (peterhoeg) [inline pkgs/applications/radio/calls/default.nix]: libhandy is a runtime dependency - should go into ```buildInputs```.

File: pkgs/applications/radio/calls/default.nix
Comment (peterhoeg) [inline pkgs/applications/radio/calls/default.nix]: runHook preCheck (and postCheck at the end)

**Review by peterhoeg — CHANGES_REQUESTED** [review]

**Comment by worldofpeace** [comment]
This and a lot of other Librem software has been packaged as part of #57614.

https://github.com/NixOS/nixpkgs/pull/57614/commits/ddee3c8a7b6fe716dc66417d88d348463be0cfdd looks pretty complete, perhaps out of date.

**Comment by craigem** [comment]
> This and a lot of other Librem software has been packaged as part of #57614.

I had not found this. I'll make sure I hook everything I'm doing back into that.

Thanks!

**Comment by craigem** [comment]
I had a read of #57614 and note that he had put it "networking" where as I had put it in "radio".

Any thoughts on the correct location?

**Comment by worldofpeace** [comment]
A calling application would be a networking application. Additionally all the applications in that directory have to relation to this one.

**Comment by craigem** [comment]
> A calling application would be a networking application. Additionally all the applications in that directory have no relation to this one.

I can see why that would be applicable, but this application is using modem manager to use operate a mobile phone radio to place phone calls across a mobile phone network.

Happy to switch this to "networking", just wanted to make it clear that this was not a VOIP application.

**Comment by worldofpeace** [comment]
> I can see why that would be applicable, but this application is using modem manager to use operate a mobile phone radio to place phone calls across a mobile phone network.

Indeed, but I think it has fallen out of fashion to refer to those things as "radio" even though it's correct.

also
>  Additionally all the applications in that directory have no relation to this one.

**Comment by craigem** [comment]
**calls** has now been moved to **networking**.

**Review by worldofpeace — CHANGES_REQUESTED** [review]
You need to call this package in `all-packages.nix`.

Additonally, can you match the code style of https://github.com/NixOS/nixpkgs/commit/ddee3c8a7b6fe716dc66417d88d348463be0cfdd and *Inputs?

Currently the dependencies are declared improperly, and we shouldn't have to reference anything under the `gnome3` attribute.

**Comment by craigem** [comment]
Thanks for your time and input @worldofpeace 

I originally based this work on how the related package, libhandy was packaged so your corrections and guidance have been helpful.

If I've understood your suggestions correctly, this last commit should be what you're looking for.

**Comment by craigem** [comment]
Finally bumped `all-packages.nix`.

File: pkgs/applications/networking/calls/default.nix
Comment (peterhoeg) [inline pkgs/applications/networking/calls/default.nix]: libhandy is already in buildInputs, so it should be removed.

File: pkgs/applications/networking/calls/default.nix
Comment (peterhoeg) [inline pkgs/applications/networking/calls/default.nix]: Are you sure this isn't a buildInput?

**Review by peterhoeg — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/calls/default.nix
Comment (jtojnar) [inline pkgs/applications/networking/calls/default.nix]:
```suggestion
    xorg.xauth
```

**Comment by jtojnar** [comment]
What the heck is this commit set:

```ShellSession
~/P/nixpkgs ((3a0f023e…)|…) $ git hub pull checkout 65362
Fetching pull/65362/head from git@github.com:NixOS/nixpkgs.git
~/P/nixpkgs ((6e99182f…)|…) $ git rebase channels/nixos-unstable
First, rewinding head to replay your work on top of it...
Applying: calls: Initial commit to package the application
Applying: Corrected licence
Applying: Corrected licence
Using index info to reconstruct a base tree...
M	pkgs/applications/radio/calls/default.nix
Falling back to patching base and 3-way merge...
Auto-merging pkgs/applications/radio/calls/default.nix
No changes -- Patch already applied.
~/P/nixpkgs ((07e4206a…)|…) $ nix-build -A calls
error: attribute 'calls' in selection path 'calls' not found
```

**Comment by lheckemann** [comment]
I've pushed a version with a cleaned-up history, upgraded to a recent version from upstream git, and tested the application as far as I could (I haven't managed to bend ModemManager to my will yet, but calls seems fine). Will merge on Monday if nobody objects, or as soon as someone approves :)

File: pkgs/applications/radio/calls/default.nix
Comment (worldofpeace) [inline pkgs/applications/radio/calls/default.nix]:
```suggestion
  buildInputs = [ libhandy gsound gtk3 libpeas gom libxml2 evolution-data-server ];
```

old aliases

File: pkgs/applications/radio/calls/default.nix
Comment (worldofpeace) [inline pkgs/applications/radio/calls/default.nix]: why is modemanager and libhandy native?

**Comment by worldofpeace** [comment]
@lheckemann This was never moved to networking https://github.com/NixOS/nixpkgs/pull/65362#issuecomment-516202918.

**Comment by worldofpeace** [comment]
@lheckemann Can we format the expression as @jtojnar https://github.com/NixOS/nixpkgs/blob/4bc5288ac9f6d99ca595433fcaf9fa4ca9fbe5a5/pkgs/applications/networking/calls/default.nix.
That's a code style we're pretty much going for. You could even push that expression manually.

**Comment by lheckemann** [comment]
Wow, that was a funky history. Rebase completely failed..? Anyway, here's my second cleanup attempt :)

File: pkgs/applications/networking/calls/default.nix
Comment (worldofpeace) [inline pkgs/applications/networking/calls/default.nix]: `unstable` in `pname`.

https://github.com/NixOS/nixpkgs/issues/68518

File: pkgs/applications/networking/calls/default.nix
Comment (worldofpeace) [reply pkgs/applications/networking/calls/default.nix]: @lheckemann ^

File: pkgs/applications/networking/calls/default.nix
Comment (worldofpeace) [inline pkgs/applications/networking/calls/default.nix]:
```suggestion
    evolution-data-server
```

File: pkgs/applications/networking/calls/default.nix
Comment (worldofpeace) [inline pkgs/applications/networking/calls/default.nix]:
```suggestion

    buildInputs = [
```

**Comment by worldofpeace** [comment]
@lheckemann `libhandy` and `modemanager` are both sill in `nativeBuildInputs`.

**Pull request merged**
